### PR TITLE
Prepare for automated data matching

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

With https://github.com/DFE-Digital/apply-for-teacher-training/pull/2451 we're going to automate the sharing of data with UCAS.

## Changes proposed in this pull request

- Remove the data export entirely from Support
- Move the relevant code into the `ucas_matching` namespace, so everything UCAS-related is colocated with after https://github.com/DFE-Digital/apply-for-teacher-training/pull/2451. 

## Guidance to review

Does it make sense to put everything related to this in a separate namespace?

## Link to Trello card

https://trello.com/c/vX5nXO2g/2331-arrange-matching-data-with-ucas

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/ucas_matching/matching_data_export.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
+module UCASMatching
+  class MatchingDataExport
+    HEADER_NAMES = {
+      apply_candidate_id: 'Apply candidate ID',
+      first_name: 'First Name',
```
Comment (duncanjbrown) [inline app/services/ucas_matching/matching_data_export.rb:5]: Should we lower-case `name` to be consistent with other two-word headers?

File: app/services/ucas_matching/matching_data_export.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      address_line_2: 'Address line 2',
+      address_line_3: 'Address line 3',
+      address_line_4: 'Address line 4',
+      application_state: 'Application state',
+      country: 'Country',
+      postcode: 'Postcode',
+      email_address: 'Email address',
+      provider_code: 'Provider code',
+      provider_name: 'Provider name',
+      phase: 'Apply stage', # <- what we call our field: how UCAS refer to it
```
Comment (duncanjbrown) [inline app/services/ucas_matching/matching_data_export.rb:18]: Late question! If there submitted-but-withdrawn applications in Apply 2 they can share a candidate ID with a "live" Apply 2 application, as in Mike E's Slack message yesterday — https://ukgovernmentdfe.slack.com/archives/CQA64BETU/p1594032853157000. These won't be uniquely identifiable from the combination of Candidate ID and phase... but perhaps that doesn't matter to UCAS?

https://www.apply-for-teacher-training.education.gov.uk/support/applications/2319
https://www.apply-for-teacher-training.education.gov.uk/support/applications/2321

File: app/services/ucas_matching/matching_data_export.rb (hunk 1 above)
Comment (tijmenb) [reply app/services/ucas_matching/matching_data_export.rb:5]: This is not a change I made, this is the file format that's agreed with UCAS.

File: app/services/ucas_matching/matching_data_export.rb (hunk 2 above)
Comment (tijmenb) [reply app/services/ucas_matching/matching_data_export.rb:18]: I don't know about this, perhaps discuss this some other time? This PR doesn't change the code at all.

Comment (duncanjbrown) [reply app/services/ucas_matching/matching_data_export.rb:18]: Sure — no intention of changing anything, just curious

**Pull request merged**